src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,55 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { Error } from '../Error';
+
+interface TodoModalProps {
+  todo: Todo;
+  onClose: () => void;
+}
+
+export const TodoModal: React.FC<TodoModalProps> = ({ todo, onClose }) => {
+  const {
+    id,
+    userId,
+    title,
+    completed,
+  } = todo;
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [user, setUser] = useState<User>();
+  const [hasError, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    setIsLoading(true);
```

Comment:
When you have code blocks it's better to have empty line to separate them
```suggestion
    setIsLoading(true);
    
```